kratos/utilities/time_discretization.h

```diff
@@ -25,47 +26,124 @@ class ProcessInfo; // forward-declaring to not having to include it here
 
 namespace TimeDiscretization {
 
-class KRATOS_API(KRATOS_CORE) BDF1
+class KRATOS_API(KRATOS_CORE) BDF
 {
 public:
-    std::array<double, 2> ComputeBDFCoefficients(double DeltaTime) const;
-    std::array<double, 2> ComputeBDFCoefficients(const ProcessInfo& rProcessInfo) const;
+
+    KRATOS_CLASS_POINTER_DEFINITION(BDF);
+
+    /**
+     * @brief Construct a new BDF object
+     * Auxiliary constructor for derived classes
+     */
+    BDF(){};
+
+    /**
+     * @brief Construct a new BDF object
+     * Constructor with time order
+     * @param TimeOrder BDF time order
+     */
+    BDF(const unsigned int TimeOrder) : mTimeOrder(TimeOrder)
+    {
+        this->SetAuxBDFPointer(TimeOrder, mpAuxBDF);
+    };
+
+    /**
+     * @brief Destroy the BDF object
+     * Destructor of the BDF class
+     */
+    virtual ~BDF(){};
+
+    /**
+     * @brief Set the Aux BDF Pointer object
+     * This method sets a pointer to one of the auxiliary BDF
+     * classes according to the user defined time order
+     * @param TimeOrder BDF scheme time order
+     * @param rpAuxBDF Pointer to the auxiliary BDF scheme
+     */
+    static void SetAuxBDFPointer(
+        const unsigned int TimeOrder,
+        unique_ptr<BDF> &rpAuxBDF);
+
+    /**
+     * @brief Return the BDF coefficients
+     * This method computes the BDF coefficients for a provided time step
+     * @param DeltaTime Time step to compute the BDF coefficients
+     * @return std::vector<double> Vector containing the computed BDF coefficients
+     */
+    virtual std::vector<double> ComputeBDFCoefficients(double DeltaTime) const;
+
+    /**
+     * @brief Return the BDF coefficients
+     * This method computes the BDF coefficients for a provided current and old time step
+     * @param DeltaTime Time step to compute the BDF coefficients
+     * @param PreviousDeltaTime Old time step to compute the BDF coefficients
+     * @return std::vector<double> Vector containing the computed BDF coefficients
+     */
+    virtual std::vector<double> ComputeBDFCoefficients(double DeltaTime, double PreviousDeltaTime) const;
+
+    /**
+     * @brief Return the BDF coefficients
+     * This method computes the BDF coefficients for the time step
+     * stored in the variable DELTA_TIME of a ProcessInfo container
+     * @param rProcessInfo ProcessInfo container with DELTA_TIME
+     * @return std::vector<double> Vector containing the computed BDF coefficients
+     */
+    virtual std::vector<double> ComputeBDFCoefficients(const ProcessInfo &rProcessInfo) const;
```

Comment:
```suggestion
    virtual std::vector<double> ComputeBDFCoefficients(const ProcessInfo& rProcessInfo) const;
```
most classes in the core do it like this, e.g. `Element`